File: Hash/leetcode/454.go
package leetcode

func fourSumCount(nums1 []int, nums2 []int, nums3 []int, nums4 []int) int {
	m := make(map[int]int)
	for _,v1 := range nums1 {
			for _, v2 := range nums2 {
					m[v1+v2]++
			} 
	}

	count := 0
	for _, v3 := range nums3 {
			for _, v4 := range nums4 {
					count += m[-v3-v4]
			}
	}
	return count
}

File: Hash/cpp/454.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        unordered_map<int, int> umap;
        for (int a : nums1) {
            for (int b : nums2) {
                umap[a+b]++;
            }
        }

        int count = 0;

        for (int c : nums3) {
            for (int d : nums4) {
                if (umap.find(0-(c+d)) != umap.end()) {
                    count += umap[0-(c+d)];
                }
            }
        }
        return count;
    }
};